lib/license_finder/cli/main.rb

```diff
@@ -175,6 +180,20 @@ def report_of(content)
       def save?
         !!config.save_file
       end
+
+      # Filter out proprietary dependencies (for license page only)
+      def filter_proprietary(dependencies)
+        return dependencies unless config.format == 'license-page'
```

Comment:
Now that's an *ugly* coupling, but it gets the job done.  How would you do this if you wanted to submit this work as a PR back to the original upstream?  I'm fine with some GSD here for now, but damn, that's ugly.

Reply:
> How would you do this if you wanted to submit this work as a PR back to the original upstream?

Filters. For example, command line parameter like `--filter=` where filter can be

(a) the name of configuration file (config file should have some format)
or
(b) some ruby code (decorator) that can be applied to `dependencies`.

Default filter can be named "Null (or dummy) filter" without filtering functionality. With decorators we can apply as many decorators as we want, for example: `--filter=proprietary,mit`. Maybe `--filter` is not a good name, `--skip-dependencies=` looks better.

Reply:
Also, filters and templates are different concepts, so here it would be also nice to introduce `--template=` parameter.